administrator/components/com_installer/Field/downloadkey.php

```diff
@@ -0,0 +1,46 @@
+<?php
+/**
+ * @package     Joomla.Platform
+ * @subpackage  Form
+ *
+ * @copyright   Copyright (C) 2005 - 2017 Open Source Matters, Inc. All rights reserved.
+ * @license     GNU General Public License version 2 or later; see LICENSE
+ */
+
+defined('JPATH_PLATFORM') or die;
+
+JFormHelper::loadFieldClass('text');
+
+/**
+ * Download Key field.
+ *
+ * @since  __DEPLOY_VERSION__
+ */
+class JFormFieldDownloadkey extends JFormFieldText
+{
+	/**
+	 * The form field type.
+	 *
+	 * @var    string
+	 * @since  __DEPLOY_VERSION_
+	 */
+	public $type = 'Downloadkey';
+
+	/**
+	 * Method to get the field input for Download key.
+	 *
+	 * @return  string  The field input.
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function getInput()
+	{
+		$extraQuery = InstallerHelper::getExtraQuery($this->form->getValue('update_site_id'));
```

Comment:
This should be done as three separate form fields in the XML, not rendering three input fields within one form field class.  This probably also means you will need extra processing in the model when the form is created and the data filled in to split the data into the correct fields.